crates/nu-command/src/filesystem/save.rs

```diff
@@ -318,10 +322,20 @@ fn stream_to_file(
     mut stream: RawStream,
     file: File,
     span: Span,
+    progress: bool,
 ) -> Result<PipelineData, ShellError> {
     let mut writer = BufWriter::new(file);
 
-    stream
+    let mut bytes_processed: u64 = 0;
+    let bytes_processed_p = &mut bytes_processed;
+    let file_total_size = stream.known_size.map(|n| (n as u64)); // type TBD
+
+    // Create the progress bar and set the style.
+    let mut bar = progress_bar::NuProgressBar::new(file_total_size);
+    bar.create_style();
+    let f_bar = bar.clone();
+
+    let r_stream = stream
```

Comment:
What do `f_` and `r_` mean here?

Reply:
Thanks! I took a closer look at this code just now, some notes:

1. `r_stream` is not a stream. It's a `Result<PipelineData, ShellError>` representing the result of processing the stream. So maybe a better name would be something like `result`?
2. If saving fails (i.e. `r_stream` is an `Err`), we still report "File successfully Saved!". Probably don't want to do that.